app/streams/publishing_api/message_adapter.rb

```diff
@@ -1,35 +1,78 @@
 module PublishingAPI
   class MessageAdapter
-    def self.to_dimension_item(*args)
-      new(*args).to_dimension_item
+    attr_reader :items
+
+    def self.to_dimension_items(*args)
+      new(*args).to_dimension_items
     end
 
     def initialize(message)
       @message = message
+      @items = []
     end
 
-    def to_dimension_item
+    def to_dimension_item(item_message)
       Dimensions::Item.new(
-        content_id: message.payload.fetch('content_id'),
-        base_path: message.payload.fetch('base_path'),
-        publishing_api_payload_version: message.payload.fetch('payload_version'),
-        document_type: message.payload.fetch('document_type'),
-        locale: message.payload['locale'],
-        title: message.payload['title'],
-        content_purpose_document_supertype: message.payload['content_purpose_document_supertype'],
-        content_purpose_supergroup: message.payload['content_purpose_supergroup'],
-        content_purpose_subgroup: message.payload['content_purpose_subgroup'],
+        content_id: item_message.payload.fetch('content_id'),
+        base_path: item_message.payload.fetch('base_path'),
+        publishing_api_payload_version: item_message.payload.fetch('payload_version'),
+        document_type: item_message.payload.fetch('document_type'),
+        locale: item_message.payload['locale'],
+        title: item_message.payload['title'],
+        content_purpose_document_supertype: item_message.payload['content_purpose_document_supertype'],
+        content_purpose_supergroup: item_message.payload['content_purpose_supergroup'],
+        content_purpose_subgroup: item_message.payload['content_purpose_subgroup'],
         first_published_at: parse_time('first_published_at'),
         primary_organisation_content_id: primary_organisation['content_id'],
         primary_organisation_title: primary_organisation['title'],
         primary_organisation_withdrawn: primary_organisation['withdrawn'],
         public_updated_at: parse_time('public_updated_at'),
-        schema_name: message.payload.fetch('schema_name'),
+        schema_name: item_message.payload.fetch('schema_name'),
         latest: true,
-        raw_json: message.payload,
+        raw_json: item_message.payload,
       )
     end
 
+    def to_dimension_items
```

Comment:
Could you please add a test for this? You might be able to dry it up afterwards as the only different between `to_dimension_item` and `to_dimension_items` is the `base_path`